src/pages-and-resources/teams/TeamSetEditor.jsx

```diff
@@ -0,0 +1,208 @@
+import messages from './messages';
+
+/**
+ * Team sizes enum
+ * @enum
+ * @type {{MIN: number, MAX: number, DEFAULT: number}}
+ */
+export const TeamSizes = {
+  DEFAULT: 50,
+  MIN: 1,
+  MAX: 500,
+};
+
+/**
+ * Team types enum
+ * @enum
+ * @type {{PRIVATE_MANAGED: string, PUBLIC_MANAGED: string, OPEN: string}}
+ */
+export const TeamSetTypes = {
+  OPEN: 'open',
+  PUBLIC_MANAGED: 'public_managed',
+  PRIVATE_MANAGED: 'private_managed',
+};
+
+// Maps a team type to its corresponding intl message
+const TeamTypeNameMessage = {
+  [TeamSetTypes.OPEN]: { label: messages.teamSetTypeOpen, description: messages.teamSetTypeOpenDescription },
+  [TeamSetTypes.PUBLIC_MANAGED]: {
+    label: messages.teamSetTypePublicManaged,
+    description: messages.teamSetTypePublicManagedDescription,
+  },
+  [TeamSetTypes.PRIVATE_MANAGED]: {
+    label: messages.teamSetTypePrivateManaged,
+    description: messages.teamSetTypePrivateManagedDescription,
+  },
+};
+
+function TeamSetEditor({
+  intl, teamSet, onDelete, onChange, onBlur, fieldNameCommonBase, errors, touched,
+}) {
+  const [isDeleting, setDeleting] = useState(false);
+  const [isOpen, setOpen] = useState(teamSet.id === null);
+  const initiateDeletion = () => setDeleting(true);
+  const cancelDeletion = () => setDeleting(false);
+
+  return (
+    <CollapsableEditor
+      // If there is no id, this is a new team, so automatically open it
+      open={isOpen}
+      onToggle={setOpen}
+      onDelete={initiateDeletion}
+      deleteAlt={intl.formatMessage(messages.deleteAlt)}
+      expandAlt={intl.formatMessage(messages.expandAlt)}
+      collapseAlt={intl.formatMessage(messages.collapseAlt)}
+      title={
+        isOpen
+          ? (
+            <div className="d-flex flex-column small flex-shrink-1 h4">
+              {intl.formatMessage(messages.configureTeamSet)}
+            </div>
+          ) : (
+            <div className="d-flex flex-column small flex-shrink-1">
+              <span>{intl.formatMessage(TeamTypeNameMessage[teamSet.type].label)}</span>
+              <span className="h4 text-truncate">{teamSet.name || '<new>'}</span>
+              <span className="text-truncate">{teamSet.description || '<new>'}</span>
+            </div>
+          )
+}
+    >
+      <TransitionReplace>
+        {isDeleting
+          ? (
+            <div className="d-flex flex-column m-4" key="isDeleting">
+              <h3>{intl.formatMessage(messages.teamSetDeleteHeading)}</h3>
+              <p>{intl.formatMessage(messages.teamSetDeleteBody)}</p>
+              <div className="d-flex flex-row justify-content-end">
+                <Button variant="muted" size="sm" onClick={() => onDelete(teamSet)}>
+                  {intl.formatMessage(messages.delete)}
+                </Button>
+                <Button variant="muted" size="sm" onClick={cancelDeletion}>
+                  {intl.formatMessage(messages.cancel)}
+                </Button>
+              </div>
```

Comment:
Delete design is not matching with Figma. Deleting function is not working properly. It has many edge cases bugs.
https://www.loom.com/share/ffd5a5ef993f4d5da93a018651ed1bc7

current design: 
<img width="775" alt="Screenshot 2021-09-10 at 5 38 51 PM" src="https://user-images.githubusercontent.com/79941147/132854437-549c409b-c075-4060-b619-881ba09dcabd.png">


Figma Design:
<img width="480" alt="Screenshot 2021-09-10 at 3 25 14 PM" src="https://user-images.githubusercontent.com/79941147/132854337-568a8344-c358-47df-b791-5794fa8bc93d.png">